PS-6055 Add support for Float and Double data

mohitj1988 opened the pull request:
With this PR, pqueryv3.0 will have support for Float and Double data types.

mohitj1988 committed:
PS-6055 Add support for Float and Double data
types columns in pqueryv3.0

rahulmalik87 commented on src/random_test.cpp:
```diff
@@ -726,22 +758,26 @@ void Table::CreateDefaultColumn() {
       /* loop untill we select some column */
       while (col_type == Column::COLUMN_MAX) {
 
-        /* columns are 4:3:2:1:1 INT:VARCHAR:CHAR:BLOB:BOOL */
```
please keep the ratio of (float+ double) less than of int. 
maybe (2:1)

mohitj1988 replied:
Fixed

rahulmalik87 commented on src/random_test.cpp:
```diff
@@ -258,6 +258,12 @@ int rand_int(int upper, int lower) {
 
+/* return random float number in the range of upper and lower */
+float rand_float(float upper, float lower) {
+  float r = (rand() / (float)RAND_MAX * (upper - 1)) + lower;
```
don't use the rand function.  use some normal distributions.

mohitj1988 replied:
Fixed

rahulmalik87 commented on src/random_test.cpp:
```diff
@@ -325,6 +339,14 @@ std::string Column::rand_value() {
     static auto rec = 100 * opt_int(INITIAL_RECORDS_IN_TABLE);
     return std::to_string(rand_int(rec));
     break;
+  case (COLUMN_TYPES::FLOAT):
```
case FLOAT and DOUBLE seems to be doing the same thing. 
so you can either combine them or see if there is a better way to get rand_float
for them.

mohitj1988 replied:
Fixed

rahulmalik87 reviewed (commented):
add column type handing in load_metadata_from_file.

mohitj1988 committed:
Addressed review comments -
- Changed rand() distribution for float and double to
normal distribution
- Removed some debug print messages from the code

rahulmalik87 commented on src/random_test.cpp:
```diff
@@ -260,8 +262,10 @@ int rand_int(int upper, int lower) {
 float rand_float(float upper, float lower) {
-  float r = (rand() / (float)RAND_MAX * (upper - 1)) + lower;
-  return r;
+  std::random_device rd;
```
we have already created the seed. Directly call uniform_real_distribution

rahulmalik87 commented on src/random_test.cpp:
```diff
@@ -340,13 +344,25 @@ std::string Column::rand_value() {
+    rec2 /= 100;
+    std::ostringstream out;
+    out << std::fixed;
+    out << std::setprecision(5) << rand_float(rec2);
```
with set precision(5). there will be very hard to find records of double in update/delete queries. you need a better division.

mohitj1988 replied:
Fixed

mohitj1988 committed:
Addressed review comments.

rahulmalik87 commented on src/random_test.cpp:
```diff
@@ -726,22 +772,26 @@ void Table::CreateDefaultColumn() {
 
-        /* columns are 4:3:2:1:1 INT:VARCHAR:CHAR:BLOB:BOOL */
-        auto prob = rand_int(10);
+        /* columns are 6:2:2:2:2:2:1 INT:FLOAT:DOUBLE:VARCHAR:CHAR:BLOB:BOOL */
```
varchar probability becomes very less . make it 4.

mohitj1988 replied:
Fixed

rahulmalik87 commented on src/random_test.cpp:
```diff
@@ -325,6 +341,26 @@ std::string Column::rand_value() {
     break;
+  case (COLUMN_TYPES::FLOAT):
+  {
+    float rec1 = opt_int(INITIAL_RECORDS_IN_TABLE);
```
use something like static auto rec = opt_int(INITIAL_RECORDS_IN_TABLE)/ 100;

mohitj1988 replied:
Fixed

rahulmalik87 commented on src/random_test.cpp:
```diff
@@ -258,6 +260,12 @@ int rand_int(int upper, int lower) {
 }
 
+/* return random float number in the range of upper and lower */
+float rand_float(float upper, float lower) {
```
you can add one more argument precision. and move logic here.

mohitj1988 committed:
Merge remote-tracking branch 'upstream/master' into PS-6055

rahulmalik87 reviewed (changes requested):
please

mohitj1988 committed:
Addressed Some more review comments

mohitj1988 committed:
Addressed leftover comments

rahulmalik87 commented on src/random_test.cpp:
```diff
@@ -258,6 +260,15 @@ int rand_int(int upper, int lower) {
+  std::uniform_real_distribution<> dis(lower, upper);
+  std::ostringstream out;
+  out << std::fixed;
+  out << std::setprecision(precision) << (float)(dis(rng)*100.0)/100.0;
```
(float)(dis(rng)*100.0)/100.0. 
can it be done with dis(rng)

mohitj1988 replied:
Fixed

mohitj1988 committed:
created new functions rand_float and rand_double.
